scripts/release-branch-forward/release-branch-forward.go

```diff
@@ -0,0 +1,109 @@
+// This tools automatically finds the latest CRI-O release branch and merges
+// the latest master branch into it. This happens only if there is no
+// tag present on the release branch.
+package main
+
+import (
+	"flag"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/release/pkg/command"
+	kgit "k8s.io/release/pkg/git"
+)
+
+const (
+	remote              = "https://github.com/cri-o/cri-o"
+	git                 = "git"
+	grep                = "grep"
+	tail                = "tail"
+	releaseBranchPrefix = "release-"
+)
+
+var dryRun bool
+
+func init() {
+	flag.BoolVar(
+		&dryRun,
+		"dry-run",
+		true,
+		"do not relly push, just do a dry run",
+	)
+	flag.Parse()
+}
+
+func main() {
+	logrus.SetFormatter(&logrus.TextFormatter{DisableTimestamp: true})
+	if err := run(); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func run() error {
+	// Get the latest release branch
+	lsRemoteHeads, err := command.
+		New(git, "ls-remote", "--sort=v:refname", "--heads", remote).
+		Pipe(grep, "-Eo", releaseBranchPrefix+".*").
+		Pipe(tail, "-1").
+		RunSilentSuccessOutput()
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve latest release branch")
+	}
+	latestReleaseBranch := lsRemoteHeads.OutputTrimNL()
+	logrus.Infof("Latest release branch: %s", latestReleaseBranch)
+
+	// Check if a release has been done on that branch
+	tagPrefix := strings.TrimPrefix(latestReleaseBranch, releaseBranchPrefix)
+	lsRemoteTags, err := command.
+		New(git, "ls-remote", "--sort=v:refname", "--tags", remote).
+		Pipe(grep, "-Eo", "v"+tagPrefix+".*").
+		RunSilentSuccessOutput()
+	if err == nil {
+		logrus.Warnf(
+			"Found existing tag(s) on release branch: %v",
+			strings.Join(strings.Fields(lsRemoteTags.OutputTrimNL()), ", "),
+		)
+		logrus.Infof("We’re all set, doing nothing")
+		return nil
+	}
+
+	// Checkout the release branch
+	repo, err := kgit.OpenRepo(".")
+	if err != nil {
+		return errors.Wrap(err, "unable to open this repository")
+	}
+	if dryRun {
+		logrus.Info("Setting repository to do only a dry-run")
```

Comment:
if you fix the other nit, "do only" -> "only do"

Reply:
Fixed as well :+1: